src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisPipelineService.java

```diff
@@ -0,0 +1,70 @@
+import de.tum.in.www1.artemis.domain.ProgrammingSubmission;
+import de.tum.in.www1.artemis.domain.iris.session.IrisChatSession;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.PyrisPipelineExecutionSettingsDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisCourseDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisUserDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.status.PyrisStageDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.status.PyrisStageStateDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.tutorChat.PyrisTutorChatPipelineExecutionDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.job.TutorChatJob;
+import de.tum.in.www1.artemis.service.iris.websocket.IrisChatWebsocketService;
+
+@Service
+@Profile("iris")
+public class PyrisPipelineService {
+
+    private final PyrisConnectorService pyrisConnectorService;
+
+    private final PyrisJobService pyrisJobService;
+
+    private final PyrisDTOService pyrisDTOService;
+
+    private final IrisChatWebsocketService irisChatWebsocketService;
+
+    @Value("${server.url}")
+    private String artemisBaseUrl;
+
+    public PyrisPipelineService(PyrisConnectorService pyrisConnectorService, PyrisJobService pyrisJobService, PyrisDTOService pyrisDTOService,
+            IrisChatWebsocketService irisChatWebsocketService) {
+        this.pyrisConnectorService = pyrisConnectorService;
+        this.pyrisJobService = pyrisJobService;
+        this.pyrisDTOService = pyrisDTOService;
+        this.irisChatWebsocketService = irisChatWebsocketService;
+    }
+
+    /**
+     * Executes the tutor chat pipeline for the given session
+     *
+     * @param variant          the variant of the pipeline
+     * @param latestSubmission the latest submission of the user
+     * @param exercise         the exercise the user is working on
+     * @param session          the chat session
+     */
+    public void executeTutorChatPipeline(String variant, Optional<ProgrammingSubmission> latestSubmission, ProgrammingExercise exercise, IrisChatSession session) {
+        var jobToken = pyrisJobService.addJob(new TutorChatJob(exercise.getCourseViaExerciseGroupOrCourseMember().getId(), exercise.getId(), session.getId()));
+        var settingsDTO = new PyrisPipelineExecutionSettingsDTO(jobToken, List.of(), artemisBaseUrl);
+        var preparingRequestStageInProgress = new PyrisStageDTO("Preparing request", 10, PyrisStageStateDTO.IN_PROGRESS, "Checking out repositories and loading data");
+        var preparingRequestStageDone = new PyrisStageDTO("Preparing request", 10, PyrisStageStateDTO.DONE, "Checking out repositories and loading data");
+        var executingPipelineStageNotStarted = new PyrisStageDTO("Executing pipeline", 30, PyrisStageStateDTO.NOT_STARTED, null);
+        var executingPipelineStageInProgress = new PyrisStageDTO("Executing pipeline", 30, PyrisStageStateDTO.IN_PROGRESS, null);
```

Comment:
This looks like a duplicate?

Reply:
Nope. One is `NOT_STARTED` and the other is `IN_PROGRESS`.
The first is reported to the client immediatley, the other when the fetching of the repos (and other data collection on Artemis side) is done, as this can take about 2 seconds. 
Since these are records, this cannot just be changed and thus we need to create two instances.